Use Gentoo-appropriate absolute paths

Original author: Alexandre Rostovtsev <tetromino@gentoo.org>
---
--- a/panels/common/Makefile.am
+++ b/panels/common/Makefile.am
@@ -9,7 +9,7 @@
 	$(LIBLANGUAGE_CFLAGS)				\
 	-DDATADIR=\""$(datadir)"\"			\
 	-DUIDIR=\""$(pkgdatadir)/ui"\"			\
-	-DLIBLOCALEDIR=\""$(prefix)/lib/locale"\"       \
+	-DLIBLOCALEDIR=\""$(libdir)/locale"\"       \
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"        \
 	-DUM_PIXMAP_DIR=\""$(pkgdatadir)/pixmaps"\"
 
--- a/panels/datetime/Makefile.am
+++ b/panels/datetime/Makefile.am
@@ -72,6 +72,7 @@
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""	\
 	-DGNOMECC_DATA_DIR="\"$(pkgdatadir)\""		\
 	-DDATADIR="\"$(uidir)\""			\
+	-DREALDATADIR="\"$(datadir)\""			\
 	$(NULL)
 
 noinst_PROGRAMS = test-timezone-gfx test-endianess test-timezone
--- a/panels/datetime/test-endianess.c
+++ b/panels/datetime/test-endianess.c
@@ -38,11 +38,11 @@
 		return 0;
 	}
 
-	dir = g_dir_open ("/usr/share/i18n/locales/", 0, NULL);
+	dir = g_dir_open (REALDATADIR "/i18n/locales/", 0, NULL);
 	if (dir == NULL) {
 		/* Try with /usr/share/locale/
 		 * https://bugzilla.gnome.org/show_bug.cgi?id=646780 */
-		dir = g_dir_open ("/usr/share/locale/", 0, NULL);
+		dir = g_dir_open (REALDATADIR "/locale/", 0, NULL);
 		if (dir == NULL) {
 			return 1;
 		}
--- a/panels/datetime/test-timezone.c
+++ b/panels/datetime/test-timezone.c
@@ -1,7 +1,7 @@
 #include <gtk/gtk.h>
 #include "cc-timezone-map.h"
 
-#define TZ_DIR "/usr/share/zoneinfo/"
+#define TZ_DIR REALDATADIR "/zoneinfo/"
 
 static GList *
 get_timezone_list (GList *tzs,
--- a/panels/datetime/tz.h
+++ b/panels/datetime/tz.h
@@ -29,9 +29,9 @@
 #include <glib.h>
 
 #ifndef __sun
-#  define TZ_DATA_FILE "/usr/share/zoneinfo/zone.tab"
+#  define TZ_DATA_FILE REALDATADIR "/zoneinfo/zone.tab"
 #else
-#  define TZ_DATA_FILE "/usr/share/lib/zoneinfo/tab/zone_sun.tab"
+#  define TZ_DATA_FILE REALDATADIR "/lib/zoneinfo/tab/zone_sun.tab"
 #endif
 
 typedef struct _TzDB TzDB;
--- a/panels/printers/Makefile.am
+++ b/panels/printers/Makefile.am
@@ -12,6 +12,9 @@
 	-DGNOMELOCALEDIR="\"$(datadir)/locale\""	\
 	-DGNOMECC_DATA_DIR="\"$(pkgdatadir)\""		\
 	-DDATADIR="\"$(uidir)\""			\
+	-DREALDATADIR="\"$(datadir)\""			\
+	-DPREFIX="\"$(prefix)\""			\
+	-DLIBEXECDIR="\"$(libexecdir)\""			\
 	$(NULL)
 
 ccpanelsdir = $(PANELS_DIR)
--- a/panels/printers/cc-printers-panel.c
+++ b/panels/printers/cc-printers-panel.c
@@ -2199,8 +2199,8 @@
 
   if (printer_name)
     {
-      const gchar  *const dirs[] = { "/usr/share/cups",
-                                     "/usr/local/share/cups",
+      const gchar  *const dirs[] = { REALDATADIR "/cups",
+                                     PREFIX "/local/share/cups",
                                      NULL };
       const gchar  *testprint[] = { "%s/data/testprint",
                                     "%s/data/testprint.ps",
--- a/panels/printers/pp-host.c
+++ b/panels/printers/pp-host.c
@@ -251,7 +251,7 @@
   data->devices->devices = NULL;
 
   argv = g_new0 (gchar *, 3);
-  argv[0] = g_strdup ("/usr/lib/cups/backend/snmp");
+  argv[0] = g_strdup (LIBEXECDIR "/cups/backend/snmp");
   argv[1] = g_strdup (priv->hostname);
 
   /* Use SNMP to get printer's informations */
--- a/panels/user-accounts/Makefile.am
+++ b/panels/user-accounts/Makefile.am
@@ -10,7 +10,7 @@
 AM_CPPFLAGS =						\
 	-DDATADIR=\""$(datadir)"\"			\
 	-DUIDIR=\""$(pkgdatadir)/ui/user-accounts"\"	\
-	-DLIBLOCALEDIR=\""$(prefix)/lib/locale"\"       \
+	-DLIBLOCALEDIR=\""$(libdir)/locale"\"       \
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"        \
 	-DUM_PIXMAP_DIR=\""$(pkgdatadir)/pixmaps"\"	\
 	-I$(srcdir)/../common/				\
-- 
1.7.8.6

